### ISSUE TITLE
electron 1.4.2 / v8 5.3 fails to compile

### ISSUE THREAD

**Issue by dimatter** [laverdet/node-fibers#311]
What am I doing wrong?
`$ HOME=~/.electron-gyp node-gyp rebuild --target=1.4.2 --arch=x64 --dist-url=https://atom.io/download/atom-shell`
I really don't understand why electron reports using node v6.5 yet I need to recompile fibers against iojs1.4.2 from 2015-02-28?!

My guess is [this](https://codereview.chromium.org/1953263002) could be related ?

```javascript
{ http_parser: '2.7.0',
  node: '6.5.0',
  v8: '5.3.332.45',
  uv: '1.9.1',
  zlib: '1.2.8',
  ares: '1.10.1-DEV',
  modules: '50',
  openssl: '1.0.2h',
  electron: '1.4.2',
  chrome: '53.0.2785.113',
  'atom-shell': '1.4.2' }
```

```
Anonymous :: fibers-test/node_modules/fibers 1 » HOME=~/.electron-gyp node-gyp rebuild --target=1.4.2 --arch=x64 --dist-url=https://atom.io/download/atom-shell &&
mkdir ./bin/darwin-x64-v8-5.3/ &&
mv build/Release/fibers.node bin/darwin-x64-v8-5.3/
gyp info it worked if it ends with ok
gyp info using node-gyp@3.4.0
gyp info using node@6.5.0 | darwin | x64
gyp info spawn /usr/local/bin/python2
gyp info spawn args [ '/Users/dimatter/.nvm/versions/node/v6.5.0/lib/node_modules/node-gyp/gyp/gyp_main.py',
gyp info spawn args   'binding.gyp',
gyp info spawn args   '-f',
gyp info spawn args   'make',
gyp info spawn args   '-I',
gyp info spawn args   '/Users/dimatter/Sandbox/fibers-test/node_modules/fibers/build/config.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/Users/dimatter/.nvm/versions/node/v6.5.0/lib/node_modules/node-gyp/addon.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/Users/dimatter/.electron-gyp/.node-gyp/iojs-1.4.2/common.gypi',
gyp info spawn args   '-Dlibrary=shared_library',
gyp info spawn args   '-Dvisibility=default',
gyp info spawn args   '-Dnode_root_dir=/Users/dimatter/.electron-gyp/.node-gyp/iojs-1.4.2',
gyp info spawn args   '-Dnode_gyp_dir=/Users/dimatter/.nvm/versions/node/v6.5.0/lib/node_modules/node-gyp',
gyp info spawn args   '-Dnode_lib_file=iojs.lib',
gyp info spawn args   '-Dmodule_root_dir=/Users/dimatter/Sandbox/fibers-test/node_modules/fibers',
gyp info spawn args   '--depth=.',
gyp info spawn args   '--no-parallel',
gyp info spawn args   '--generator-output',
gyp info spawn args   'build',
gyp info spawn args   '-Goutput_dir=.' ]
gyp info spawn make
gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
  CXX(target) Release/obj.target/fibers/src/fibers.o
../src/fibers.cc:49:34: error: no template named 'WeakCallbackData' in namespace
      'v8'; did you mean 'WeakCallbackInfo'?
        void WeakCallbackShim(const v8::WeakCallbackData<T, P>& data) {
                                    ~~~~^~~~~~~~~~~~~~~~
                                        WeakCallbackInfo
/Users/dimatter/.electron-gyp/.node-gyp/iojs-1.4.2/deps/v8/include/v8.h:6799:16: note:
      'WeakCallbackInfo' declared here
  friend class WeakCallbackInfo;
               ^
../src/fibers.cc:49:34: error: too many template arguments for class template
      'WeakCallbackInfo'
        void WeakCallbackShim(const v8::WeakCallbackData<T, P>& data) {
                                        ^                   ~~
/Users/dimatter/.electron-gyp/.node-gyp/iojs-1.4.2/deps/v8/include/v8.h:6799:16: note:
      template is declared here
  friend class WeakCallbackInfo;
               ^
../src/fibers.cc:102:18: warning: 'NewFromOneByte' is deprecated
      [-Wdeprecated-declarations]
                return String::NewFromOneByte(isolate, (const uint8_t*)string);
                               ^
/Users/dimatter/.electron-gyp/.node-gyp/iojs-1.4.2/deps/v8/include/v8.h:2314:21: note:
      'NewFromOneByte' has been explicitly marked deprecated here
      Local<String> NewFromOneByte(Isolate* isolate, const uint8_t* data,
                    ^
../src/fibers.cc:106:18: warning: 'NewFromOneByte' is deprecated
      [-Wdeprecated-declarations]
                return String::NewFromOneByte(isolate, (const uint8_t*)string);
                               ^
/Users/dimatter/.electron-gyp/.node-gyp/iojs-1.4.2/deps/v8/include/v8.h:2314:21: note:
      'NewFromOneByte' has been explicitly marked deprecated here
      Local<String> NewFromOneByte(Isolate* isolate, const uint8_t* data,
                    ^
../src/fibers.cc:423:80: warning: 'NewInstance' is deprecated
      [-Wdeprecated-declarations]
  ...uni::Return(uni::Deref(Isolate::GetCurrent(), tmpl)->GetFunction()->NewI...
                                                                         ^
/Users/dimatter/.electron-gyp/.node-gyp/iojs-1.4.2/deps/v8/include/v8.h:3283:31: note:
      'NewInstance' has been explicitly marked deprecated here
                Local<Object> NewInstance(int argc, Local<Value> argv[]) const);
                              ^
../src/fibers.cc:611:14: warning: 'TryCatch' is deprecated
      [-Wdeprecated-declarations]
                                TryCatch try_catch;
                                         ^
/Users/dimatter/.electron-gyp/.node-gyp/iojs-1.4.2/deps/v8/include/v8.h:6932:40: note:
      'TryCatch' has been explicitly marked deprecated here
  V8_DEPRECATED("Use isolate version", TryCatch());
                                       ^
../src/fibers.cc:731:34: warning: 'ToNumber' is deprecated
      [-Wdeprecated-declarations]
                        Coroutine::pool_size = value->ToNumber()->Value();
                                                      ^
/Users/dimatter/.electron-gyp/.node-gyp/iojs-1.4.2/deps/v8/include/v8.h:8298:22: note:
      'ToNumber' has been explicitly marked deprecated here
Local<Number> Value::ToNumber() const {
                     ^
../src/fibers.cc:785:8: warning: 'SetAccessor' is deprecated
      [-Wdeprecated-declarations]
                        fn->SetAccessor(uni::NewLatin1Symbol(isolate, "c...
                            ^
/Users/dimatter/.electron-gyp/.node-gyp/iojs-1.4.2/deps/v8/include/v8.h:2751:22: note:
      'SetAccessor' has been explicitly marked deprecated here
                bool SetAccessor(Local<String> name,
                     ^
../src/fibers.cc:786:8: warning: 'SetAccessor' is deprecated
      [-Wdeprecated-declarations]
                        fn->SetAccessor(uni::NewLatin1Symbol(isolate, "p...
                            ^
/Users/dimatter/.electron-gyp/.node-gyp/iojs-1.4.2/deps/v8/include/v8.h:2751:22: note:
      'SetAccessor' has been explicitly marked deprecated here
                bool SetAccessor(Local<String> name,
                     ^
../src/fibers.cc:787:8: warning: 'SetAccessor' is deprecated
      [-Wdeprecated-declarations]
                        fn->SetAccessor(uni::NewLatin1Symbol(isolate, "f...
                            ^
/Users/dimatter/.electron-gyp/.node-gyp/iojs-1.4.2/deps/v8/include/v8.h:2751:22: note:
      'SetAccessor' has been explicitly marked deprecated here
                bool SetAccessor(Local<String> name,
                     ^
../src/fibers.cc:55:10: error: no matching member function for call to 'SetWeak'
                handle.SetWeak(val, WeakCallbackShim<F>);
                ~~~~~~~^~~~~~~
../src/fibers.cc:341:9: note: in instantiation of function template
      specialization 'uni::MakeWeak<&Fiber::WeakCallback, v8::Object, void>'
      requested here
                        uni::MakeWeak<WeakCallback>(isolate, handle, (vo...
                             ^
/Users/dimatter/.electron-gyp/.node-gyp/iojs-1.4.2/deps/v8/include/v8.h:544:18: note:
      candidate function template not viable: requires 3 arguments, but 2 were
      provided
  V8_INLINE void SetWeak(P* parameter,
                 ^
/Users/dimatter/.electron-gyp/.node-gyp/iojs-1.4.2/deps/v8/include/v8.h:555:18: note:
      candidate function not viable: requires 0 arguments, but 2 were provided
  V8_INLINE void SetWeak();
                 ^
8 warnings and 3 errors generated.
make: *** [Release/obj.target/fibers/src/fibers.o] Error 1
gyp ERR! build error
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/Users/dimatter/.nvm/versions/node/v6.5.0/lib/node_modules/node-gyp/lib/build.js:276:23)
gyp ERR! stack     at emitTwo (events.js:106:13)
gyp ERR! stack     at ChildProcess.emit (events.js:191:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:215:12)
gyp ERR! System Darwin 15.5.0
gyp ERR! command "/Users/dimatter/.nvm/versions/node/v6.5.0/bin/node" "/Users/dimatter/.nvm/versions/node/v6.5.0/bin/node-gyp" "rebuild" "--target=1.4.2" "--arch=x64" "--dist-url=https://atom.io/download/atom-shell"
gyp ERR! cwd /Users/dimatter/Sandbox/fibers-test/node_modules/fibers
gyp ERR! node -v v6.5.0
gyp ERR! node-gyp -v v3.4.0
gyp ERR! not ok
```

**Comment by unboundmusic** [comment]
Same issue here. +1

**Comment by rosen-vladimirov** [comment]
@laverdet we had similar issues with older versions of electron. In fact the electron itself has a fork of node and v8 and they modify some core node functions. Also they pull v8 commits directly from the upstream, so it's really difficult to support electron. Our final solution was to use node binary and spawn it from electron's node process. In the new process we have a real, stable and tested node environment and that's where we load the fibers. This works like a charm in a long term. Before that, each update of electron was a real pain for us.

**Comment by laverdet** [comment]
Fixed in 8d7e4ffeb5151ade2ef32455080fbb2ffc226e13

It's in npm now as 1.0.15.
